build.cake

```diff
@@ -378,36 +401,98 @@ Task("Install-Unity-Windows").Does(() => {
     }
 }).OnError(HandleError);
 
-// Downloading UWP IL2CPP dependencies.
+
+async Task<SourcePackageDependencyInfo> ResolvePackage(DependencyInfoResource dependencyResource, PackageIdentity identity) 
+{
+    var cacheContext = new SourceCacheContext();
+    var logger = NullLogger.Instance;
+
+    var package = await dependencyResource.ResolvePackage(identity, Netstandard2FrameworkName, cacheContext, logger, CancellationToken.None);
+    if (package != null) 
+    {
+        return package;
+    }
+    Information($"Haven't found UWP version for {identity.Id} framework for {Netstandard2FrameworkName.ToString()}. Searching for {Netstandard2FrameworkName.ToString()}");
+    package = await dependencyResource.ResolvePackage(identity, UwpFrameworkName, cacheContext, logger, CancellationToken.None);
+    if (package != null) 
+    {
+        return package;
+    }
+    if (package == null) 
+    {
+        throw new Exception($"Package {identity.Id} not found!");
+    }
+    return null;
+}
+
+async Task GetRecursiveDependenciesCore(DependencyInfoResource dependencyResource, string id, NuGetVersion version, List<NugetDependency> dependencies, HashSet<int> uniqueIds, bool addDependencies = true)
+{
+    Information($"GetRecursiveDependenciesCore {id} version {version.Version.ToString()}");
+    var package = await ResolvePackage(dependencyResource, new PackageIdentity(id, version));
+    dependencies.Add(new NugetDependency(package.Id, NuGetVersion.Parse(package.Version.ToNormalizedString())));
+    var hashPackageId = GetStringHash(package.Id);
+    if (!uniqueIds.Contains(hashPackageId))
+    {
+        uniqueIds.Add(hashPackageId);
+        var uri = package.DownloadUri.ToString();
+        Information("Downloading " + package.Id + " from " + uri);
+        DownloadFile (uri, ExternalsFolder + package.Id + NuPkgExtension);
+        if (!addDependencies) 
+        {
+            return;
+        }
+        foreach (var dependency in package.Dependencies) 
+        {
+            if (IgnoreNuGetDependencies.Contains(dependency.Id) || dependency.Id.StartsWith("System")) 
+            {
+                continue;
+            }
+            var hashDepId = GetStringHash(dependency.Id);
+            if (!uniqueIds.Contains(hashDepId))
+            {
+                await GetRecursiveDependenciesCore(dependencyResource, dependency.Id, NuGetVersion.Parse(dependency.VersionRange.MinVersion.ToNormalizedString()), dependencies, uniqueIds, addDependencies);
+            }
+        }
+    } 
+}
+
+private int GetStringHash(string input) 
+{
+    using (System.Security.Cryptography.HashAlgorithm algorithm = System.Security.Cryptography.SHA256.Create()) 
+    {
+        return BitConverter.ToInt32(algorithm.ComputeHash(Encoding.UTF8.GetBytes(input)), 0);
+    }
+}
+
+private DependencyInfoResource GetDefaultDependencyResource() 
+{
+    var packageSource = "https://api.nuget.org/v3/index.json";
```

Comment:
Use built in const

Reply:
I wasn't able to make use of [CAKE_NUGET_SOURCE](https://cakebuild.net/docs/fundamentals/default-configuration-values), so I just moved it to the top as a constant.